Library/Homebrew/formulary.rb

```diff
@@ -392,6 +396,102 @@ def klass(flags:, ignore_errors:)
     end
   end
 
+  # Load formulae from the API.
+  class FormulaAPILoader < FormulaLoader
+    def initialize(name)
+      super name, Formulary.core_path(name)
```

Comment:
We do the same thing (that is, having the `path` be set to the hypothetical tap path) when loading from a bottle. I think I'm leaning toward not making a change here since this directly affects what's returned by `Formula#path`, and there are too many references to this to thoroughly check to see what would happen if this was `nil` suddenly. Although it's a bit odd, I don't think it's causing any problems to have this set in this way.